astropy/coordinates/baseframe.py

```diff
@@ -1356,8 +1353,11 @@ def _frameattr_equiv(left_fattr, right_fattr):
             return False
 
         left_is_repr = isinstance(left_fattr, r.BaseRepresentationOrDifferential)
-        right_is_repr = isinstance(right_fattr, r.BaseRepresentationOrDifferential)
-        if left_is_repr and right_is_repr:
+        if left_is_repr is not isinstance(
```

Comment:
Still confuses me a bit, but it is logical and a quick way out.

Reply:
> Would it be better if the code used `!=` instead of `is not`?

Probably would throw me off a bit less... I was also wondering about `^`... But in the end this gets too nitpicky to have all CI run again for it - you should be allowed to use that `isinstance` can only return `True` or `False`. I'm happy to just get this in!

Reply:
An advantage of using `^` or `!=` would be that this condition would fit on a single line. And if I make a new commit then the code coverage check might work properly. But if you are happy with the way this is then you should merge.

Reply:
Sure, the shortening makes it worth it (I'm not worried about the coverage).

Reply:
I had to rebase anyways because of a conflict in `.ruff.toml`.